crates/bevy_ecs/src/schedule/executor/mod.rs

```diff
@@ -73,8 +75,79 @@ impl SystemSchedule {
             system_dependents: Vec::new(),
             sets_with_conditions_of_systems: Vec::new(),
             systems_in_sets_with_conditions: Vec::new(),
+            systems_with_stepping_enabled: FixedBitSet::new(),
+            step_state: StepState::RunAll,
         }
     }
+
+    /// Return the index of the next system to be run if this schedule is
+    /// stepped
+    pub fn stepping_next_system_index(&self) -> Option<usize> {
+        let index = match self.step_state {
+            StepState::RunAll => None,
+            StepState::Wait(i) | StepState::Next(i) | StepState::Remaining(i) => Some(i),
+        }?;
+
+        self.find_stepping_system(index)
+    }
+
+    /// Called by [`Executor::run`] to get the list of systems to be skipped
+    /// due to stepping; this method also updates the `step_state` to prepare
+    /// for the next call.
+    pub fn step(&mut self) -> Option<FixedBitSet> {
```

Comment:
For this `FixedBitSet` the code currently does an allocation.  I can switch this to use a `FixedBitSet` allocated with the `SystemSchedule` if this is a performance problem.  It only happens when we're taking a step, so we're supposed to be slow here.